### DISCUSSION TITLE
Nodes regression on a multi-graph database

### DISCUSSION THREAD

**Discussion by vmecaens** [danielegrattarola/spektral#377] [Q&A]
Hello to all,

I've been using this library for a week, I'm still having a little trouble with it!

I'm trying to make a GNN for a regression task on the labels of nodes. I have a database made of graphs with a number of different nodes. I'm trying to predict the labels of the nodes (3 labels /nodes), so I want to output a matrix of size: [n_nodes * n_labels].

The idea is to have a part of all the graphs for training and the other one for validation and testing, I use for that the DisjointLoader with the option node_level=True.

I have only found examples that process one graph for training and validation and that use masks on the nodes. Do you have an example of a graph that would work for such a task?

I tried with the ready-to-use GeneralGNN model (recommended for node level prediction tasks) but it only runs when I have a batch size of 1, and when I evaluate the trained model it returns a vector of size [n_labels=3] and not [n_nodes * n_labels]...

I will be curious to understand this, thanks in advance!

**Comment by rmrmg** [comment]
i found this after few days of fighting with different problems and various settings. Do you have any working example of regression with DisjointLoader?

**Comment by danielegrattarola** [comment]
For node-level prediction with GeneralGNN you should set `pool=None` so that the global pooling is not applied. 

I don't have an example but it should be pretty similar to any other example if you set `node_level=True` in the loader. Alternatively, you can build your inputs manually, you don't have to use a loader to use the models.

Also sorry that I'm replying 1 year later but I had completely missed this issue :/
